system/include/webgl/webgl1_ext.h

```diff
@@ -370,9 +469,86 @@ WEBGL_APICALL void GL_APIENTRY glMultiDrawElementsInstancedWEBGL(GLenum mode,
 #define GL_RG16_SNORM_EXT 0x8F99
 #define GL_RGB16_SNORM_EXT 0x8F9A
 #define GL_RGBA16_SNORM_EXT 0x8F9B
+// <no functions exposed>
 #endif /* EMSCRIPTEN_GL_EXT_texture_norm16 */
 
 // EMSCRIPTEN_explicit_uniform_location
-#ifndef GL_MAX_UNIFORM_LOCATIONS
+// https://github.com/emscripten-core/emscripten/blob/main/docs/EMSCRIPTEN_explicit_uniform_location.txt
+#ifndef EMSCRIPTEN_explicit_uniform_location
+#define EMSCRIPTEN_explicit_uniform_location 1
+// To enable: link with -sGL_EXPLICIT_UNIFORM_LOCATION=1
 #define GL_MAX_UNIFORM_LOCATIONS          0x826E
+// <no functions exposed>
 #endif
+
+// EMSCRIPTEN_explicit_uniform_binding
+// https://github.com/emscripten-core/emscripten/blob/main/docs/EMSCRIPTEN_explicit_uniform_binding.txt
+#ifndef EMSCRIPTEN_explicit_uniform_binding
+#define EMSCRIPTEN_explicit_uniform_binding 1
+// To enable: link with -sGL_EXPLICIT_UNIFORM_BINDING=1
+// <no functions or defines exposed>
+#endif
+
+// 50. https://registry.khronos.org/webgl/extensions/EXT_polygon_offset_clamp/
+#ifndef EMSCRIPTEN_GL_EXT_polygon_offset_clamp
+#define EMSCRIPTEN_GL_EXT_polygon_offset_clamp 1
+// To enable: call
+EM_BOOL emscripten_webgl_enable_EXT_polygon_offset_clamp(EMSCRIPTEN_WEBGL_CONTEXT_HANDLE context);
+// or link with -sGL_SUPPORT_SIMPLE_ENABLE_EXTENSIONS=1 and
+// call emscripten_webgl_enable_extension(ctx, "EXT_polygon_offset_clamp");
+#define GL_POLYGON_OFFSET_CLAMP_EXT 0x8E1B
+WEBGL_APICALL void GL_APIENTRY emscripten_glPolygonOffsetClampEXT(GLfloat factor, GLfloat units, GLfloat clamp);
+WEBGL_APICALL void GL_APIENTRY glPolygonOffsetClampEXT(GLfloat factor, GLfloat units, GLfloat clamp);
+#endif
+
+// 51. https://registry.khronos.org/webgl/extensions/EXT_clip_control/
+#ifndef EMSCRIPTEN_GL_EXT_clip_control
+#define EMSCRIPTEN_GL_EXT_clip_control 1
+// To enable: call
+EM_BOOL emscripten_webgl_enable_EXT_clip_control(EMSCRIPTEN_WEBGL_CONTEXT_HANDLE context);
+// or link with -sGL_SUPPORT_SIMPLE_ENABLE_EXTENSIONS=1 and
+// call emscripten_webgl_enable_extension(ctx, "EXT_clip_control");
+#define GL_LOWER_LEFT_EXT          0x8CA1
+#define GL_UPPER_LEFT_EXT          0x8CA2
+#define GL_NEGATIVE_ONE_TO_ONE_EXT 0x935E
+#define GL_ZERO_TO_ONE_EXT         0x935F
+#define GL_CLIP_ORIGIN_EXT         0x935C
+#define GL_CLIP_DEPTH_MODE_EXT     0x935D
+WEBGL_APICALL void GL_APIENTRY emscripten_glClipControlEXT(GLenum origin, GLenum depth);
+WEBGL_APICALL void GL_APIENTRY glClipControlEXT(GLenum origin, GLenum depth);
+#endif
+
+// 52. https://registry.khronos.org/webgl/extensions/EXT_depth_clamp/
```

Comment:
Dies these numbers have some meaning?

Reply:
Yes, they're part of the extension specification.

> ## Number
> WebGL extension \#52

Reply:
(oops didn't finish writing)
It's not a very important number but it's a fixed number and I think it's in the order that the extensions were drafted. https://registry.khronos.org/webgl/extensions/